includes/class-wc-stripe-helper.php

```
@@ -1105,6 +1105,73 @@ public static function has_cart_or_checkout_on_current_page() {
+	 * @return bool True if adaptive pricing is supported for the current checkout, false otherwise.
+	 * @since 10.5.0
+	 */
+	public static function is_adaptive_pricing_supported(): bool {
+
+		// False if checkout session feature flag is disabled.
+		if ( ! WC_Stripe_Feature_Flags::is_checkout_sessions_available() ) {
+			return false;
+		}
+
+		// False if adaptive pricing option is disabled.
+		if ( 'yes' !== WC_Stripe_Helper::get_settings( null, 'adaptive_pricing' ) ) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Helper' in method 'is_adaptive_pricing_supported'. (undefined)

(StaticAccess)